### PR TITLE
Piff 114 moments

### PR DESCRIPTION
**Pull request by eacharles**
Probably need to iterate a bit to get full coverage.

### CODE REVIEW

File: piff/util.py [hunk 1]
```diff
@@ -382,3 +382,267 @@ def calculateSNR(image, weight):
...
+
+    .. math::
+
+        r^2 &\equiv du^2 + dv^2 \\
+        M_22 &= \sum W(u,v) I(u,v) r^4
+        M_33 &= \sum W(u,v) I(u,v) r^6
+        M_44 &= \sum W(u,v) I(u,v) r^8
+        M_22n &= M_22/M_11^2
+        M_33n &= M_33/M_11^3
+        M_44n &= M_44/M_11^4
```
[the remark this answers is not in the document]
Comment (eacharles) [reply piff/util.py]: Ok, done, but note that utils.py doesn't actually show up in the doc.

File: tests/test_moments.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+        # moments
+        moments = calculate_moments(star=noiseless_stars[i],errors=True, third_order=True, fourth_order=True, radial=True)
+        moments_noise = calculate_moments(star=noisy_stars[i],errors=True, third_order=True, fourth_order=True, radial=True)
+
+        if test_options:
+            moments_34 = calculate_moments(star=noiseless_stars[i], errors=True, third_order=True, fourth_order=True, radial=False)
+            mask_34 = [True, True, True, True, True, True, True, True, True, True, True, True, True, True, True, False, False, False,
+                       True, True, True, True, True, True, True, True, True, True, True, True, True, True, True, False, False, False]
+            np.testing.assert_equal(np.array(moments)[mask_34], np.array(moments_34))
```
[the remark this answers is not in the document]
Comment (eacharles) [reply tests/test_moments.py:164]: Hmm.  I think we have different ideas of "intuitive", to me it is nice to see visually which elements are being used and which are not.   With the other two tests the values we care about aren't continuous.  This way each test is clearly pulling out the elements that it needs.

File: tests/test_moments.py [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                       True, True, True, True, True, True, True, True, True, True, False, False, False, False, False, True, True, True]
+            np.testing.assert_equal(np.array(moments)[mask_3r], np.array(moments_3r))
+
+            moments_4r = calculate_moments(star=noiseless_stars[i], errors=True, third_order=False, fourth_order=True, radial=True)
+            mask_4r = [True, True, True, True, True, True, False, False, False, False, True, True, True, True, True, True, True, True,
+                       True, True, True, True, True, True, False, False, False, False, True, True, True, True, True, True, True, True]
+            np.testing.assert_equal(np.array(moments)[mask_4r], np.array(moments_4r))
+        
+        if test_mask:
+            copy_star = piff.Star(noisy_star.data, noisy_star.fit)
```
[the remark this answers is not in the document]
Comment (eacharles) [reply tests/test_moments.py]: Star doesn't actually have a copy method.  If I add one 

def copy(self):
   return copy.deepcopy(self) 

This works fine

File: piff/util.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmjarvis) [reply piff/util.py]: ... should probably add that then, no?

File: tests/test_moments.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rmjarvis) [reply tests/test_moments.py:162]: Fair enough.   For me, arithmetic with small numbers is easier than scanning a long list of True and False's to see which items are different.  I accept that for people with better vision the opposite may be the case.  However, I'd request that you please add inline comments that highlight which elements are False in each case.  Like
```
# No radial, skip items 16-18 of each
```
to help those of us with poor eyesight.

File: tests/test_moments.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rmjarvis) [reply tests/test_moments.py]: Sorry.  My mistake.  I thought we had one.  I'm pretty sure we don't want `deepcopy` here though.  Just `copy`.  We normally want to leave it as shallow as possible.

File: piff/util.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (eacharles) [reply piff/util.py]: Ok, I did the simplest thing and added a util.rst with a call to automodule piff.util to the doc.   

I fixed up a couple of sphinx warnings in util.py

File: tests/test_moments.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (eacharles) [reply tests/test_moments.py]: That's doesn't work with the current code.   It fails unless I use this line in test_moments.py:

`noisy_star = copy.deepcopy(noiseless_star)`

If I leave that line as a shallow copy the variance of the moments distributions are all zero.    Offhand it isn't clear to my why that should fail.

[the remark this answers is not in the document]
Comment (rmjarvis) [reply tests/test_moments.py]: OK.  Strange.  I thought the line you had here would have been equivalent to a shallow copy.  Just copying the pointers to data and fit.  I thought that was basically what `copy.copy` did -- just shallow copy all the attributes.

**Comment by rmjarvis** [comment]
OK, I switched it to return a dict for the moments, and if requested a second dict with the same keys for the variances.  I think that's a nicer API than returning a long opaque list of values.  The values in the dict are ordered, so you can turn it back to a long list if you want, with the same ordering that we currently have.  But I suspect user code will be significantly more legible using the key names for access.

**Comment by rmjarvis** [comment]
I also looked into getting the ellipticity terms correct, but they quickly got out of hand as lots of high order moments became relevant (M42, M24, M04, M40, etc.).  So I think we should be content that the variances are pretty good for round stars, but maybe not so great when the ellipticity is more than a few percent.

**Pull request merged**